src/main/kotlin/org/danilopianini/plagiarismdetector/analyzer/technique/tokenization/java/JavaPreprocessor.kt

```diff
@@ -0,0 +1,18 @@
+package org.danilopianini.plagiarismdetector.analyzer.technique.tokenization.java
+
+import com.github.javaparser.ast.CompilationUnit
+import com.github.javaparser.ast.ImportDeclaration
+import org.danilopianini.plagiarismdetector.analyzer.StepHandler
+
+/**
+ * A java source file preprocessor, which removes noisy information
+ * (import and packages declarations) that can affect the overall detection
+ * accuracy of the system.
+ */
+class JavaPreprocessor : StepHandler<CompilationUnit, CompilationUnit> {
+    override fun invoke(input: CompilationUnit): CompilationUnit {
+        input.removePackageDeclaration()
+        input.findAll(ImportDeclaration::class.java).forEach(input::remove)
+        return input
+    }
```

Comment:
Prefer immutability, do not modify the input. Make `CompilationUnit` immutable. I want to prevent components to fire back at us when we'll parallelize

Reply:
`CompilationUnit` is a third library class. I can clone the input and work on the cloned object.